### PR TITLE
Created ugly_numbers.py in Python/maths

### PR DESCRIPTION
**Pull request by Kush1101**
### **Describe your change:**

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Comment by Kush1101** [comment]
@cclauss  Please review it

File: maths/ugly_numbers.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    """
+    Returns the nth ugly number.
+    >>> ugly_numbers(100)
+    1536
+    >>> ugly_numbers(1)
+    1
+    >>> ugly_numbers(20)
+    36
+    """
+    ugly_nums = [0] * (n)
```
Comment (cclauss) [inline maths/ugly_numbers.py:22]: It is a Python antipattern to create and initialize a list before filling it out.  Is there a way to grow this array as the algorithm progresses.  If so, is that faster or slower than the current method?

**Comment by cclauss** [comment]
I do not need the `@cclauss Please review it` messages.  I will review PRs when I have the time/inclination.  If you are not ready for a review then [open your PR in draft mode](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/about-pull-requests#draft-pull-requests).

File: maths/ugly_numbers.py (hunk 1 above)
Comment (Kush1101) [reply maths/ugly_numbers.py:22]: I dosen't have an impact on time. But I can change it like
```Python
ugly_nums = [1]
...
...
ugly_nums.append(next_num)
```

**Comment by Kush1101** [comment]
> I do not need the `@cclauss Please review it` messages.
> I will review PRs when I have the time/inclination. If you are not ready for a review then [open your PR in draft mode](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/about-pull-requests#draft-pull-requests).

Ok sir. I will be careful next time.

File: maths/ugly_numbers.py (hunk 1 above)
Comment (Kush1101) [reply maths/ugly_numbers.py]: I made this change.

File: maths/ugly_numbers.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+def ugly_numbers(n: int) -> int:
+    """
+    Returns the nth ugly number.
+    >>> ugly_numbers(100)
+    1536
+    >>> ugly_numbers(1)
+    1
+    >>> ugly_numbers(20)
+    36
```
Comment (cclauss) [inline maths/ugly_numbers.py:20]: Please do not just test the "happy path"...  Please add tests for ugly_numbers(-5) and ugly_numbers(5.5).

Comment (Kush1101) [reply maths/ugly_numbers.py:20]: I wrote the tests

**Review by cclauss — APPROVED** [review]
LGTM

**Pull request merged**